container-registry/container-analysis/sample.rb

```diff
@@ -0,0 +1,346 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def create_note note_id:, project_id:
+  # [START containeranalysis_create_note]
+  # note_id    = "A user-specified identifier for the note"
+  # project_id = "Your Google Cloud project ID"
+
+  require "grafeas"
+
+  # Initialize the client
+  client = Grafeas.new
+
+  formatted_parent = Grafeas::V1::GrafeasClient.project_path project_id
+  note = {
+    vulnerability: {
+      details: [
+        affected_cpe_uri:     "your-uri-here",
+        affected_package:     "your-package-here",
+        min_affected_version: { kind: :MINIMUM },
+        fixed_version:        { kind: :MAXIMUM }
+      ]
+    }
+  }
+  response = client.create_note formatted_parent, note_id, note
+  puts response.name
+  ## [END containeranalysis_create_note]
+  response
+end
+
+def delete_note note_id:, project_id:
+  # [START containeranalysis_delete_note]
+  # note_id    = "The identifier for the note to delete"
+  # project_id = "The Google Cloud project ID of the note to delete"
+
+  require "grafeas"
+
+  # Initialize the client
+  client = Grafeas.new
+
+  formatted_parent = Grafeas::V1::GrafeasClient.note_path project_id, note_id
+  client.delete_note formatted_parent
+  # [END containeranalysis_delete_note]
+end
+
+def create_occurrence resource_url:, note_id:, occurrence_project:, note_project:
+  # [START containeranalysis_create_occurrence]
+  # resource_url       = "The URL of the resource associated with the
+  #                       occurrence. eg https://gcr.io/project/image@sha256:1
+  # note_id            = "The identifier of the note associated with the
+  #                       occurrence"
+  # occurrence_project = "The Google Cloud project ID for the new occurrence"
+  # note_project       = "The Google Cloud project ID of the associated note"
+
+  require "grafeas"
+
+  # Initialize the client
+  client = Grafeas.new
+  formatted_note = Grafeas::V1::GrafeasClient.note_path note_project, note_id
+  formatted_project = Grafeas::V1::GrafeasClient.project_path occurrence_project
+
+  occurrence = {
+    note_name:     formatted_note,
+    resource_uri:  resource_url,
+    vulnerability: {
+      package_issue: [
+        affected_cpe_uri:     "your-uri-here:",
+        affected_package:     "your-package-here",
+        min_affected_version: { kind: :MINIMUM },
+        fixed_version:        { kind: :MAXIMUM }
+      ]
+    }
+  }
+
+  response = client.create_occurrence formatted_project, occurrence
+  puts response.name
+  # [END containeranalysis_create_occurrence]
+  response
+end
+
+def delete_occurrence occurrence_id:, project_id:
+  # [START containeranalysis_delete_occurrence]
+  # occurrence_id = "The API-generated identifier associated with the
+  #                  occurrence"
```

Comment:
I think some of these comments could be put on a single line?

Reply:
Not without going over 80 characters. Isn't that a hard limit for our ruby samples?

Reply:
Yes, but perhaps we could simplify the wording in some comments. Example:

> The API-generated id associated with this occurrence